packages/react/src/time-picker/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { type DefaultComboboxOption } from '../combobox/utils';
+import { formatTime, type TimeFormat } from '../time-input/utils';
+
+export function filterOptions<Option extends DefaultComboboxOption>(
+	options: Option[],
+	inputValue: string | undefined = '',
+	selectedItems?: Option[]
+) {
+	const sanitizedInputValue =
+		inputValue?.toLowerCase().replace(/\W/g, '') ?? '';
+
+	return options.filter(function filterOption(option) {
```

Comment:
One thing I noticed in Storybook was that when I type `12:0` I get a list that includes `am` and `pm`. When I add the extra `0`, i.e. `12:00`, then the `am` options disappear. I can add `a` to get to `am`, i.e. `12:00 a`.

With `12:0`
![image](https://github.com/agriculturegovau/agds-next/assets/43688949/ccf0134c-d299-4c2c-b7ef-af336f0ad2ed)


With `12:00`
![image](https://github.com/agriculturegovau/agds-next/assets/43688949/f770e87e-6f4d-4094-a606-8a7b0f7a573b)

Reply:
Yeah, I'm not sure how to fix that 😔

Reply:
@stowball I've pushed up some changes to your branch that fixes this bug